modules/openapi-generator/src/main/resources/typescript-axios/apiInner.mustache

```diff
@@ -131,17 +161,43 @@ export const {{classname}}AxiosParamCreator = function (configuration?: Configur
                 {{/isDate}}
                 {{/isDateTime}}
             }
+            {{/useSingleRequestParameter}}
+            {{#useSingleRequestParameter}}
+            if (requestParameters.{{paramName}} !== undefined) {
+                {{#isDateTime}}
+                localVarQueryParameter['{{baseName}}'] = (requestParameters.{{paramName}} as any instanceof Date) ?
+                    (requestParameters.{{paramName}} as any).toISOString() : requestParameters.{{paramName}};
+                {{/isDateTime}}
+                {{^isDateTime}}
+                {{#isDate}}
+                localVarQueryParameter['{{baseName}}'] = (requestParameters.{{paramName}} as any instanceof Date) ?
+                    (requestParameters.{{paramName}} as any).toISOString().substr(0,10) : requestParameters.{{paramName}};
+                {{/isDate}}
+                {{^isDate}}
+                localVarQueryParameter['{{baseName}}'] = requestParameters.{{paramName}};
+                {{/isDate}}
+                {{/isDateTime}}
+            }
+            {{/useSingleRequestParameter}}            
             {{/isListContainer}}
-
     {{/queryParams}}
     {{#headerParams}}
             {{#isListContainer}}
-            if ({{paramName}}) {
-                let mapped = {{paramName}}.map(value => (<any>"{{{dataType}}}" !== "Array<string>") ? JSON.stringify(value) : (value || ""));
-                localVarHeaderParameter['{{baseName}}'] = mapped.join(COLLECTION_FORMATS["{{collectionFormat}}"]);
-            }
+                {{^useSingleRequestParameter}}
+                    if ({{paramName}}) {
+                    let mapped = {{paramName}}.map(value => (<any>"{{{dataType}}}" !== "Array<string>") ? JSON.stringify(value) : (value || ""));
+                    localVarHeaderParameter['{{baseName}}'] = mapped.join(COLLECTION_FORMATS["{{collectionFormat}}"]);
+                    }
+                {{/useSingleRequestParameter}}
+                {{#useSingleRequestParameter}}
+                    if (requestParameters.{{paramName}}) {
+                    let mapped = requestParameters.{{paramName}}.map(value => (<any>"{{{dataType}}}" !== "Array<string>") ? JSON.stringify(value) : (value || ""));
+                    localVarHeaderParameter['{{baseName}}'] = mapped.join(COLLECTION_FORMATS["{{collectionFormat}}"]);
+                    }
+                {{/useSingleRequestParameter}}   
```

Comment:
@erdembas I wonder if you can revise the code indentation in the above code block. Thanks.

Reply:
@wing328 this block indent edited.